packages/plugins/other/visitor-plugin-common/src/client-side-base-visitor.ts

```diff
@@ -425,16 +429,21 @@ export class ClientSideBaseVisitor<
           if (this._omitDefinitions === true) {
             return `{${`"__meta__":${JSON.stringify(meta)},`.slice(0, -1)}}`;
           }
-          return `new TypedDocumentString(\`${doc}\`, ${JSON.stringify(meta)})`;
+          return `new TypedDocumentString(\`${escapedLiteralContents}\`, ${JSON.stringify(meta)})`;
         }
       }
 
-      return `new TypedDocumentString(\`${doc}\`)`;
+      return `new TypedDocumentString(\`${escapedLiteralContents}\`)`;
     }
 
+    const escapedLiteralContentsPrefix = asTemplateLiteral(docLiteralContents).slice(1, -1);
```

Comment:
I'd love to further extend this for honoring `optimizeDocumentNode` (but doing so introduces whitespace changes that break tests):
```suggestion
    const optimizedDocPrefix = this.config.optimizeDocumentNode
      ? print(optimizeDocumentNode(gqlTag([docLiteralContents])))
      : docLiteralContents;
    const escapedLiteralContentsPrefix = asTemplateLiteral(optimizedDocPrefix).slice(1, -1);
```